```

[2025-09-16T12:56:49.390Z] ======================================================================

[2025-09-16T12:56:49.390Z] ERROR: test_cache (tests.data.test_gdsdataset.TestDataset)

[2025-09-16T12:56:49.390Z] testing no inplace change to the hashed item

[2025-09-16T12:56:49.390Z] ----------------------------------------------------------------------

[2025-09-16T12:56:49.390Z] Traceback (most recent call last):

[2025-09-16T12:56:49.390Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/tests/data/test_gdsdataset.py", line 93, in test_cache

[2025-09-16T12:56:49.390Z]     assert_allclose(ds[0], ds1[0], type_test=False)

[2025-09-16T12:56:49.390Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/data/dataset.py", line 109, in __getitem__

[2025-09-16T12:56:49.390Z]     return self._transform(index)

[2025-09-16T12:56:49.390Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/data/dataset.py", line 418, in _transform

[2025-09-16T12:56:49.390Z]     pre_random_item = self._cachecheck(self.data[index])

[2025-09-16T12:56:49.390Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/data/dataset.py", line 1611, in _cachecheck

[2025-09-16T12:56:49.390Z]     _meta = self._load_meta_cache(meta_hash_file_name=f"{hashfile.name}-meta")

[2025-09-16T12:56:49.390Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/data/dataset.py", line 1691, in _load_meta_cache

[2025-09-16T12:56:49.390Z]     return torch.load(self.cache_dir / meta_hash_file_name, weights_only=True)

[2025-09-16T12:56:49.390Z]   File "/usr/local/lib/python3.10/dist-packages/torch/serialization.py", line 1274, in load

[2025-09-16T12:56:49.390Z]     raise pickle.UnpicklingError(_get_wo_message(str(e))) from None

[2025-09-16T12:56:49.390Z] _pickle.UnpicklingError: Weights only load failed. Re-running `torch.load` with `weights_only` set to `False` will likely succeed, but it can result in arbitrary code execution. Do it only if you got the file from a trusted source.

[2025-09-16T12:56:49.390Z] Please file an issue with the following so that we can make `weights_only=True` compatible with your use case: WeightsUnpickler error: Unsupported operand 149

[2025-09-16T12:56:49.390Z] 

[2025-09-16T12:56:49.390Z] Check the documentation of torch.load to learn more about types accepted by default with weights_only 
```

cc @ericspod

--- a/tests/data/test_gdsdataset.py
+++ b/tests/data/test_gdsdataset.py
@@ -86,7 +86,8 @@ class TestDataset(unittest.TestCase):
                     cache_dir=tempdir,
                     device=0,
                     pickle_module="pickle",
-                    pickle_protocol=pickle.HIGHEST_PROTOCOL,
+                    # TODO: was pickle.HIGHEST_PROTOCOL but this wasn't compatible with torch.load, need to improve compatibility
+                    pickle_protocol=torch.serialization.DEFAULT_PROTOCOL,
                 )
                 assert_allclose(items[0], p(np.arange(0, np.prod(shape)).reshape(shape)))
                 ds1 = GDSDataset(items, transform=_InplaceXform(), cache_dir=tempdir, device=0)

--- a/tests/data/test_gdsdataset.py
+++ b/tests/data/test_gdsdataset.py
@@ -12,7 +12,6 @@
 from __future__ import annotations
 
 import os
-import pickle
 import tempfile
 import unittest
 

--- a/tests/data/test_persistentdataset.py
+++ b/tests/data/test_persistentdataset.py
@@ -12,12 +12,12 @@
 from __future__ import annotations
 
 import os
-import pickle
 import tempfile
 import unittest
 
 import nibabel as nib
 import numpy as np
+import torch
 from parameterized import parameterized
 
 from monai.data import PersistentDataset, json_hashing
@@ -66,7 +66,8 @@ class TestDataset(unittest.TestCase):
                 transform=_InplaceXform(),
                 cache_dir=tempdir,
                 pickle_module="pickle",
-                pickle_protocol=pickle.HIGHEST_PROTOCOL,
+                # TODO: was pickle.HIGHEST_PROTOCOL but this wasn't compatible with torch.load, need to improve compatibility
+                pickle_protocol=torch.serialization.DEFAULT_PROTOCOL,
             )
             self.assertEqual(items, [[[]], [[0]], [[0, 1]], [[0, 1, 2]], [[0, 1, 2, 3]]])
             ds1 = PersistentDataset(items, transform=_InplaceXform(), cache_dir=tempdir)

